core/src/execution/mod.rs

```diff
@@ -197,6 +197,7 @@ impl<N: NetworkSpec, R: ExecutionRpc<N>> ExecutionClient<N, R> {
 
         let tx_hashes = block.transactions.hashes();
 
+        // TODO: replace this with rpc.get_block_receipts?
```

Comment:
@ncitron; thoughts on this?

Using `getBlockReceipts` instead of `getTransactionReceipt` would significantly reduce the number of HTTP calls made to the RPC.

Reply:
https://github.com/ethereum/execution-apis/pull/438

erigon, [geth](https://github.com/ethereum/go-ethereum/pull/27702), [besu](https://github.com/hyperledger/besu/pull/5771), [nimbus](https://github.com/status-im/nimbus-eth1/issues/1945), [reth](https://github.com/paradigmxyz/reth/pull/3321), 

[infura](https://docs.infura.io/api/networks/blast/json-rpc-methods/eth_getblockreceipts), [quicknode](https://www.quicknode.com/docs/ethereum/eth_getBlockReceipts), [alchemy](https://docs.alchemy.com/reference/eth-getblockreceipts), [chainstack](https://docs.chainstack.com/reference/ethereum-getblockreceipts)

yep, seems to be widely supported.

Reply:
Yeah in that case lets do it. Thanks for finding this optimization!